### PR TITLE
Using the new app.constant format

### PR DESCRIPTION
**Pull request by HEskandari**

### CODE REVIEW

File: src/Particular.PlatformSample/AppLauncher.cs [hunk 1]
```diff
@@ -55,19 +56,8 @@ public void Monitoring(int port, string logPath, string transportPath)
...
+        public void ServicePulse(int port)
         {
-            var config = GetResource("Particular.configs.app.constants.js");
-
-            config = config.Replace("{Version}", typeof(ServicePulse).Assembly.GetCustomAttribute<ServicePulseVersionAttribute>().Version);
-            config = config.Replace("{DefaultRoute}", defaultRoute ?? "/dashboard");
-            config = config.Replace("{ServiceControlPort}", serviceControlPort.ToString());
-            config = config.Replace("{MonitoringPort}", monitoringPort.ToString());
-
-            var configPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"platform\servicepulse\js\app.constants.js");
```
Comment (DavidBoike) [inline src/Particular.PlatformSample/AppLauncher.cs]: The `app.constants.js` file itself should also be removed from the project.

File: src/Particular.PlatformSample/Particular.PlatformSample.csproj [hunk 2]
```diff
@@ -15,7 +15,7 @@
     <PackageReference Include="Particular.Packaging" Version="0.2.1" PrivateAssets="all" />
     <PackageReference Include="Particular.PlatformSample.ServiceControl" Version="3.7.0" PrivateAssets="none" />
     <PackageReference Include="Particular.PlatformSample.ServiceControl.Monitoring" Version="2.3.0" PrivateAssets="none" />
-    <PackageReference Include="Particular.PlatformSample.ServicePulse" Version="1.19.0" PrivateAssets="none" />
+    <PackageReference Include="Particular.PlatformSample.ServicePulse" Version="1.19.1-pullrequest0746-0089" PrivateAssets="none" />
```
Comment (DavidBoike) [inline src/Particular.PlatformSample/Particular.PlatformSample.csproj]: Won't the real package (and version of ServicePulse) be a new minor?

**Comment by bording** [comment]
Is there some place where I can get more context around why these changes are being made?

**Comment by DavidBoike** [comment]
@bording https://github.com/Particular/DeveloperExperience/issues/83 - Pulse is going to allow configuring the various URLs through the web UI, so the app.constants.js file will go away.

File: src/Particular.PlatformSample/AppLauncher.cs (hunk 1 above)
Comment (bording) [reply src/Particular.PlatformSample/AppLauncher.cs]: There are likely even more changes that need to be made, for example see the stuff in the project file.

**Comment by bording** [comment]
@DavidBoike How is the version that used to be in there being handled now?

**Comment by DavidBoike** [comment]
Don't know, good question for @HEskandari & @mauroservienti.

**Comment by mauroservienti** [comment]
It’s still in there, the `app.constants.js` file will still be needed even though is not mandatory for connections configuration as now URLs can be passed to Pulse using query string. Anyway don’t remove the file, there are a couple more changes we might make to the way it’s used and to it’s format.

**Comment by bording** [comment]
@mauroservienti In that case then the changes in this PR are incorrect, since it's completely wiped out setting values in `app.constants.js`.

**Comment by mauroservienti** [comment]
I’m not sure I understand as I have zero knowledge of the package internals. The constants file is embedded in ServicePulse as all the other JavaScript/HTML files. Till 1.19 the file must be extracted (or available standalone) so that values can be tweaked to configure connections. In the next version that’s not needed anymore since connections configuration can be passed to ServicePulse through the URL query string. So in essence ServicePulse still needs that file, but it doesn’t need to be extracted or available on disk for the Package to be able to successfully connect.

**Comment by DavidBoike** [comment]
The `version` attribute of app.constants.js is the actual ServicePulse version - that's **build artifact stuff** and this happens to be the way we're marshalling that information into JavaScriptLand, where the rest of the stuff is "config file stuff" you're changing the paradigm on.

So that still has to be provided - @mauroservienti are you saying that the Particular.PlatformSample.ServicePulse package will now contain the app.constants.js file, containing the version according to the build contained in that package?

**Comment by bording** [comment]
> So that still has to be provided - @mauroservienti are you saying that the Particular.PlatformSample.ServicePulse package will now contain the app.constants.js file, containing the version according to the build contained in that package?

@DavidBoike I took another look at all of this, I believe that is actually the case, which I didn't realize when I made my initial comment.

If that is true, there there is even more stuff that needs to be removed as part of this PR.

**Comment by DavidBoike** [comment]
Not sure exactly where the work is on SP itself, but would it be possible to leave those settings in app.constants.js intact? That would potentially require _less change_, and allow demos (including monitoring demo) to open ServicePulse without requiring the ugliest URL ever.

**Comment by mauroservienti** [comment]
> are you saying that the Particular.PlatformSample.ServicePulse package will now contain the app.constants.js file, containing the version according to the build contained in that package?

I’m not sure, that’s something I need to check. ServicePulse when installed via the installer writes that information at the end of the installation process. That information should be really written by TeamCity 😕

Based on my guessing probably we still need the file in the package but with just the version constant. I’ll investigate that part as I don’t know the details.

**Comment by mauroservienti** [comment]
> Not sure exactly where the work is on SP itself, but would it be possible to leave those settings in app.constants.js intact? That would potentially require less change, and allow demos (including monitoring demo) to open ServicePulse without requiring the ugliest URL ever.

I raised a PR a few hours ago to enable exactly that 😊 I read minds as second job 😆 the goal is to make so that ServicePulse can use both what’s in the configuration file and/or what’s in query string.

**Comment by HEskandari** [comment]
@mauroservienti So if we will still support app.constants.js, we don't need to change anything here, correct?

**Comment by mauroservienti** [comment]
If https://github.com/Particular/ServicePulse/pull/747 gets in and works as expected (if only I could not depend on TC to build the installer I would have already tested it) then yes, no changes are required here. If only 😆

**Comment by mauroservienti** [comment]
https://github.com/Particular/ServicePulse/pull/747 has been mergewd into the current work we're doing in Pulse to allow connections configuration. With https://github.com/Particular/ServicePulse/pull/747 the `app.constants.js` file is now supported again, even though the format changed. I'll work with @HEskandari to get this PR going with the new changes.

File: src/Particular.PlatformSample/Particular.PlatformSample.csproj (hunk 2 above)
Comment (mauroservienti) [reply src/Particular.PlatformSample/Particular.PlatformSample.csproj]: it'll be `1.20`

File: src/Particular.PlatformSample/AppLauncher.cs (hunk 1 above)
Comment (HEskandari) [reply src/Particular.PlatformSample/AppLauncher.cs]: @bording What are the other changes that you had in mind?

File: src/Particular.PlatformSample/Particular.PlatformSample.csproj (hunk 2 above)
Comment (HEskandari) [reply src/Particular.PlatformSample/Particular.PlatformSample.csproj]: @DavidBoike I'll swap out the pre-release with the RTM package once it is officially released.

**Comment by HEskandari** [comment]
@DavidBoike @bording With the latest changes in ServicePulse, we no longer need to pass the connections via URLs. The PR has been updated to reflect that. Basically, the app.constant.js format has been changed and nothing else.

Please feel free to review again.

File: src/Particular.PlatformSample/PlatformLauncher.cs [hunk 3]
```diff
@@ -88,7 +88,6 @@ public static void Launch(bool showPlatformToolConsoleOutput = false, string ser
                     var serviceControlApiUrl = $"http://localhost:{controlPort}/api";
                     Console.WriteLine();
                     Console.WriteLine($"ServiceControl API can now be accessed via: {serviceControlApiUrl}");
-
```
Comment (bording) [inline src/Particular.PlatformSample/PlatformLauncher.cs]: nitpick: Can you add this line back so there the is no diff in this file?

Comment (HEskandari) [reply src/Particular.PlatformSample/PlatformLauncher.cs]: Sure can!

**Comment by HEskandari** [comment]
@mauroservienti Updated to the released package. Please give your green seal of approval so it can be merged.

**Pull request merged**